javascript/0080-remove-duplicates-from-sorted-array-ii.js

```diff
@@ -1,22 +1,26 @@
-/**
+/*
```

Comment:
The original code was following the common 2-pointer algorithm approach. This change has moved from that pattern and made it more difficult to retain.

Here’s a modified version for the 2-pointer algorithm.

```js
/**
 * 2-pointer
 * Time O(N) | Space O(1)
 * https://leetcode.com/problems/remove-duplicates-from-sorted-array-ii
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = (nums) => {
    const isEdgeCase = (nums.length < 2)
    if (isEdgeCase) return nums.length;

    let [ left, right ] = [ 2, 2 ];

    while (right < nums.length) {
        const isEqual = (nums[left - 2] === nums[right]);
        if (!isEqual) {
            nums[left] = nums[right];
            left += 1;
        }

        right += 1;
    }

    return left;
};
```